modules/dream2nix/WIP-python-pdm/default.nix

```diff
@@ -108,12 +108,23 @@ in {
     interpreter = config.deps.python.withPackages (
       ps:
         config.mkDerivation.propagatedBuildInputs
+        ++ [
+          (import ./editable.nix {
+            inherit (config.deps) runCommand python;
+            inherit (pyproject) pyproject;
+            inherit libpyproject;
+            inherit lib;
+            editablePackageSources = {
+              "${pyproject.pyproject.project.name}" = /${builtins.getEnv "PWD"}/src;
```

Comment:
* We probably shouldn't hardcode  `/src` here, as not all project use a "src layout", see e.g. https://backend.pdm-project.org/build_config/.

* Using `builtins.getEnv "PWD"` should work fine in impure mode, but maybe we could show a nicer error such as "Error: editable installs are only supported in impure mode." or so?